library/dokku_letsencrypt.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# from ansible.module_utils.basic import *
+from ansible.module_utils.basic import AnsibleModule
+import subprocess
+
+DOCUMENTATION = """
+---
+module: dokku_letsencrypt
+short_description: Enable or disable the letsencrypt plugin for a dokku app
+options:
+  app:
+    description:
+      - The name of the app
+    required: True
+    default: null
+    aliases: []
+  state:
+    description:
+      - The state of the letsencrypt plugin
+    required: False
+    default: present
+    choices: [ "present", "absent" ]
+    aliases: []
+author: Gavin Ballard
+requirements:
+  - the `dokku-letsencrypt` plugin
+"""
+
+EXAMPLES = """
+- name: Enable the letsencrypt plugin
+  dokku_letsencrypt:
+    app: hello-world
+
+- name: Disable the letsencrypt plugin
+  dokku_letsencrypt:
+    app: hello-world
+    state: absent
+"""
+
+
+def force_list(var):
+    if isinstance(var, list):
+        return var
+    return list(var)
+
+
+def subprocess_check_output(command, split="\n"):
+    error = None
+    output = []
+    try:
+        output = subprocess.check_output(command, shell=True)
+        if isinstance(output, bytes):
+            output = output.decode("utf-8")
+        output = str(output).rstrip("\n")
+        if split is None:
+            return output, error
+
+        output = output.split(split)
+        output = force_list(filter(None, output))
+        output = [o.strip() for o in output]
+    except subprocess.CalledProcessError as e:
+        error = str(e)
+    return output, error
+
+
+def dokku_letsencrypt(data):
```

Comment:
Maybe
```suggestion
def dokku_letsencrypt_list(data):
```
?

Reply:
I was wondering about this method name, but it does only return a single value (`"0"` if Let's Encrypt is not enabled for the given app, `"1"` if it is) rather than a list. This library method is very heavily based on the existing ones (especially [dokku_proxy.py](https://github.com/discolabs/ansible-dokku/blob/e3e43f377292047a63ef08665840f49f1cd9484c/library/dokku_proxy.py), as that is also a binary on/off method, and the patterns seems to be this method essentially asking "is this enabled for the given app?"

Perhaps updating this to be `dokku_letsencrypt_enabled` and returning `True` or `False` would make sense?

Reply:
> Perhaps updating this to be `dokku_letsencrypt_enabled` and returning `True` or `False` would make sense?

Sounds good to me.

I leave the decision up to you.